### ISSUE TITLE
False Positive input "go-version-file" is not defined in action "actions/setup-go@v3"

### ISSUE THREAD

**Issue by bflad** [rhysd/actionlint#139]
## Description

Given the following workflow:

```yaml
# Continuous integration handling for GitHub Actions workflows
name: ci-github-actions

on:
  pull_request:
    paths:
      - .github/workflows/*.yml

permissions:
  contents: read

jobs:
  actionlint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-go@v3
        with:
          go-version-file: 'go.mod'
      - run: go install github.com/rhysd/actionlint/cmd/actionlint@latest
      - run: actionlint

```

`actionlint` will current throw the following false positive error:

```text
.github/workflows/ci-github-actions.yml:19:11: input "go-version-file" is not defined in action "actions/setup-go@v3". available inputs are "check-latest", "go-version", "token" [action]
   |
19 |           go-version-file: 'go.mod'
   |           ^~~~~~~~~~~~~~~~
```

Earlier today, `actions/setup-go@v3` was updated to include the new input parameter, which is present on the tag:

- https://github.com/actions/setup-go/releases/tag/v3.1.0
- https://github.com/actions/setup-go/blob/v3/action.yml
